batch/src/main/java/org/openmrs/analytics/JdbcFetchUtil.java

```diff
@@ -166,22 +166,33 @@ public Integer fetchMaxId(String tableName) throws SQLException {
 		}
 		return rangeMap;
 	}
-	
-	public Map<String, String> createFhirReverseMap(String searchString, String tableFhirMapPath) throws IOException {
+	/**
+	 * creates a Map<String, List<String>> of table names to a list of coresponding Resources passed to the searchList argument
+	 */
+	public Map<String, List<String>> createFhirReverseMap(String searchString, String tableFhirMapPath) throws IOException {
 		Gson gson = new Gson();
 		Path pathToFile = Paths.get(tableFhirMapPath);
 		try (Reader reader = Files.newBufferedReader(pathToFile.toAbsolutePath(), StandardCharsets.UTF_8)) {
 			GeneralConfiguration generalConfiguration = gson.fromJson(reader, GeneralConfiguration.class);
 			Map<String, EventConfiguration> tableToFhirMap = generalConfiguration.getEventConfigurations();
 			String[] searchList = searchString.split(",");
-			Map<String, String> reverseMap = new HashMap<String, String>();
+			Map<String, List<String>> reverseMap = new HashMap<String, List<String>>();
 			for (Map.Entry<String, EventConfiguration> entry : tableToFhirMap.entrySet()) {
 				Map<String, String> linkTemplate = entry.getValue().getLinkTemplates();
 				for (String search : searchList) {
 					if (linkTemplate.containsKey("fhir") && linkTemplate.get("fhir") != null) {
 						String[] resourceName = linkTemplate.get("fhir").split("/");
+						List<String> resources;
```

Comment:
This could be moved to where it is assigned (I know there are two places but it is better to keep the scope of each variable to the minimum needed).
```suggestion
```